pyomo/opt/tests/blackbox/test_problem.py

```diff
@@ -139,26 +144,36 @@ def do_setup(self,flag):
         self.problem=TestProblem1()
         self.rproblem=RealProblem3()
 
+    def compare_xml(self, file1, file2):
+        with open(file1, 'r') as f1, open(file2, 'r') as f2:
+            f1_contents = f1.read().replace('>', '> ').replace('</', ' </').split()
+            f2_contents = f2.read().replace('>', '> ').replace('</', ' </').split()
+            for item1, item2 in zip(f1_contents, f2_contents):
```

Comment:
This should probably use `itertools.zip_longest` so that differing length files cause errors.